samples/snippets/publisher.py

```diff
@@ -179,39 +175,93 @@ def callback(f):
+    from concurrent import futures
     from google.cloud import pubsub_v1
 
     # TODO(developer)
     # project_id = "your-project-id"
     # topic_id = "your-topic-id"
 
-    # Configure the batch to publish as soon as there is ten messages,
-    # one kilobyte of data, or one second has passed.
+    # Configure the batch to publish as soon as there are 10 messages
+    # or 1 KiB of data, or 1 second has passed.
     batch_settings = pubsub_v1.types.BatchSettings(
         max_messages=10,  # default 100
-        max_bytes=1024,  # default 1 MB
+        max_bytes=1024,  # default 1 MiB
         max_latency=1,  # default 10 ms
     )
     publisher = pubsub_v1.PublisherClient(batch_settings)
     topic_path = publisher.topic_path(project_id, topic_id)
+    publish_futures = []
 
     # Resolve the publish future in a separate thread.
     def callback(future):
         message_id = future.result()
         print(message_id)
 
     for n in range(1, 10):
-        data = "Message number {}".format(n)
+        data = f"Message number {n}"
         # Data must be a bytestring
         data = data.encode("utf-8")
-        future = publisher.publish(topic_path, data)
+        publish_future = publisher.publish(topic_path, data)
         # Non-blocking. Allow the publisher client to batch multiple messages.
-        future.add_done_callback(callback)
+        publish_future.add_done_callback(callback)
+        publish_futures.append(publish_future)
+
+    futures.wait(publish_futures, return_when=futures.ALL_COMPLETED)
 
     print(f"Published messages with batch settings to {topic_path}.")
     # [END pubsub_publisher_batch_settings]
 
 
+def publish_messages_with_flow_control_settings(project_id, topic_id):
+    """Publishes messages to a Pub/Sub topic with flow control settings."""
+    # [START pubsub_publisher_flow_control]
+    from concurrent import futures
+    from google.cloud import pubsub_v1
+    from google.cloud.pubsub_v1.types import (
+        LimitExceededBehavior,
+        PublisherOptions,
+        PublishFlowControl,
+    )
+
+    # TODO(developer)
+    # project_id = "your-project-id"
+    # topic_id = "your-topic-id"
+
+    # Configure how many messages the publisher client can hold in memory
+    # and what to do when messages exceed the limit.
+    flow_control_settings = PublishFlowControl(
+        message_limit=100,  # 100 messages
+        byte_limit=10 * 1024 * 1024,  # 10 MiB
+        limit_exceeded_behavior=LimitExceededBehavior.BLOCK,
+    )
+    publisher = pubsub_v1.PublisherClient(
+        publisher_options=PublisherOptions(flow_control=flow_control_settings)
+    )
+    topic_path = publisher.topic_path(project_id, topic_id)
+    publish_futures = []
+
+    # Resolve the publish future in a separate thread.
+    def callback(publish_future):
+        message_id = publish_future.result()
+        print(message_id)
+
+    # Publish 1000 messages in quick succession to trigger flow control.
```

Comment:
I wouldn't say "to trigger flow control." It's possible that flow control will be triggered, but it's also entirely possible it won't be if messages get out quickly enough.  Either way, the fact that flow control was triggered isn't something that will be easily visible to the user. Maybe say "Rapidly publishing 1000 messages in a loop may be constrained by flow control."